sql/core/benchmarks/CollationBenchmark-results.txt

```diff
@@ -1,54 +1,79 @@
+UTF8_BINARY                                           3196           3197           2          0.0       31955.7       1.2X
+UNICODE_CI                                           19038          19043           7          0.0      190383.4       0.2X
 
-OpenJDK 64-Bit Server VM 17.0.11+9-LTS on Linux 6.5.0-1018-azure
+OpenJDK 64-Bit Server VM 17.0.11+9-LTS on Linux 6.5.0-1021-azure
 AMD EPYC 7763 64-Core Processor
 collation unit benchmarks - hashFunction:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-UTF8_BINARY_LCASE                                  7347           7349           3          0.0       73467.1       1.0X
-UNICODE                                           73462          73608         206          0.0      734623.2       0.1X
-UTF8_BINARY                                        5775           5815          57          0.0       57746.0       1.3X
-UNICODE_CI                                        57543          57619         108          0.0      575425.2       0.1X
+UTF8_BINARY_LCASE                                  6914           6921          10          0.0       69135.3       1.0X
+UNICODE                                           67702          67724          31          0.0      677019.3       0.1X
+UTF8_BINARY                                        5330           5341          15          0.0       53296.6       1.3X
+UNICODE_CI                                        65340          65342           3          0.0      653395.9       0.1X
 
-OpenJDK 64-Bit Server VM 17.0.11+9-LTS on Linux 6.5.0-1018-azure
+OpenJDK 64-Bit Server VM 17.0.11+9-LTS on Linux 6.5.0-1021-azure
 AMD EPYC 7763 64-Core Processor
 collation unit benchmarks - contains:     Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-UTF8_BINARY_LCASE                                 15415          15424          13          0.0      154147.1       1.0X
-UNICODE                                            8091           8108          25          0.0       80907.9       1.9X
-UTF8_BINARY                                        8964           8979          21          0.0       89643.5       1.7X
-UNICODE_CI                                       469123         474822        8060          0.0     4691227.7       0.0X
+UTF8_BINARY_LCASE                                116495         116514          26          0.0     1164948.6       1.0X
+UNICODE                                           52216          52232          22          0.0      522164.7       2.2X
+UTF8_BINARY                                        8520           8522           3          0.0       85196.9      13.7X
```

Comment:
quite the regression.

Reply:
I'm a bit lost, please remind me:

- are these benchmarks results referring to the worst-case scenario (aka. all different strings, so groupMapReduce has a lot of work to do)
- what are we comparing it to? meaning, what are the "old" results here in the table - is it the "old" implementation of Mode? (i.e. the one that was collation-unaware, meaning it treated all strings as UTF8_BINARY)

Reply:
@uros-db The regression we are looking at here is for the `contains` function.

Instructions for running the benchmark seem to indicate you should copy over the whole report generated, rather then just the new results. If you want I can keep everything else the same. But it seems valuable to always copy over the most recent run of benchmark for all benchmarks to keep a record of results.... 

![image](https://github.com/apache/spark/assets/31429832/cbffb6c5-86fd-4dbc-837b-5079572f811a)

Reply:
(ignore the highlighting in the screenshot -- that is just how the documentation is formatted.)

Reply:
ah I see, we're refactoring some expressions at the moment, so this regression is only temporary

I'd say let's only focus on the results for Mode